Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/ui/BooksFragment.kt

```diff
@@ -0,0 +1,100 @@
+package com.florencenjeri.readinglist.ui
+
+import android.os.Bundle
+import android.view.*
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.florencenjeri.readinglist.BooksFragmentDirections
+import com.florencenjeri.readinglist.R
+import com.florencenjeri.readinglist.ReadingListApplication
+import com.florencenjeri.readinglist.database.BooksDatabase
+import com.florencenjeri.readinglist.database.BooksRepository
+import com.florencenjeri.readinglist.model.Books
+import kotlinx.android.synthetic.main.books_fragment.*
+import kotlinx.android.synthetic.main.books_fragment.view.*
+
+class BooksFragment : Fragment(), BooksAdapter.BooksListClickListener {
+
+    private lateinit var booksViewModel: BooksViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.books_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+        val booksDao =
```

Comment:
same here, repetitiveness

Reply:
Also, I'd refrain from using the same ViewModel, in two completely different screens.

I'd have a BooksViewModel & BookDetailsViewModel